main/NotificationSystem.cpp

```diff
@@ -135,14 +135,8 @@ bool CNotificationSystem::Unregister(CNotificationObserver* pNotifier)
 	if (m_notifiers.size() > 0)
 	{
 		boost::unique_lock<boost::shared_mutex> lock(m_mutex);
-		for (size_t i = 0; i < m_notifiers.size(); i++)
-		{
-			if (m_notifiers[i] == pNotifier)
-			{
-				m_notifiers.erase(m_notifiers.begin() + i);
-				return true;
-			}
-		}
+		m_notifiers.erase(std::remove_if(m_notifiers.begin(), m_notifiers.end(), [=](CNotificationObserver *notifier) { return notifier == pNotifier; }), m_notifiers.end());
 	}
-	return false;
+
+	return std::any_of(m_notifiers.begin(), m_notifiers.end(), [=](CNotificationObserver *notifier) { return notifier == pNotifier; });
 }
```

Comment:
I think with the two changes it is actually more code, and not very readable for beginners

Reply:
There's no size difference that I can see in the binary. I think both GCC and clang are smart enough to figure out that the algos are over the same range.

Reply:
Removed anyway.